### PR TITLE
Fix FuzzedDataProvider crash on float length

### PR DESCRIPTION
**Pull request by br-lewis**
This adds checks to all functions in `FuzzedDataProvider` that accept a parameter that could be used as the number of bytes to pull from the fuzzed data. If any length value is not a whole number, the function will throw an error to prevent `dataPtr` from ending up a non-whole number. I chose this over having it do a `Math.floor` and continuing to make clear what's going on.

The crashes we were seeing were due to `consumeNumberInRange` returning a non-integer value, `consumeString` would accept that value as the length of the string to get and would set `this.dataPtr` and `this._remainingBytes` as a direct calculation based off of the length given. I'm not sure what `this.data.toString` does with the floating value but it successfully returned something so this call would work. The next call to `consumeNumberInRange` would end up trying to index `this.data` with `this.dataPtr + <some other number>` and the end result was `this.data[<a non-integer value>]` which JS will deal with by returning `undefined`.

### CODE REVIEW

File: packages/core/FuzzedDataProvider.ts [hunk 1]
```diff
@@ -14,6 +14,15 @@
  * limitations under the License.
  */
 
+export class FloatLengthError extends Error {
+	constructor() {
+		super();
+		this.name = "FLOAT_LENGTH_ERROR";
+		this.cause = "non-integer length given";
```
Comment (bertschneider) [inline packages/core/FuzzedDataProvider.ts:25]: `cause` is used to pass along caught errors and can probably be removed here.

File: packages/core/FuzzedDataProvider.ts [hunk 2]
```diff
@@ -89,6 +98,9 @@ export class FuzzedDataProvider {
 	 * @returns an integral
 	 */
 	consumeIntegral(maxNumBytes: number, isSigned = false): number {
+		if (!Number.isInteger(maxNumBytes)) {
```
[the remark this answers is not in the document]
Comment (br-lewis) [reply packages/core/FuzzedDataProvider.ts:105]: Yeah I wasn't too sure about whether to do only the public API or also internals but you're right that checking at the point of entry is the most consistent without requiring us to constantly check.

File: packages/core/FuzzedDataProvider.ts [hunk 3]
```diff
@@ -14,6 +14,15 @@
  * limitations under the License.
  */
 
+export class FloatLengthError extends Error {
```
[the remark this answers is not in the document]
Comment (br-lewis) [reply packages/core/FuzzedDataProvider.ts:21]: Good catch, I missed the one in `pickValues` and made those functions private.

**Review by bertschneider — APPROVED** [review]
Looks good! :+1: 

Could you please squash the three commits together, as they belong to one functional change.

**Pull request merged**